extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +124,194 @@ export class Transport {
+   */
+  static hasCrossDomainFrame(frameUrl) {
+    return hasOwn(Transport.crossDomainFrames, frameUrl);
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!HTMLDocument} ampDoc  The document node of the parent page
+   * @param {!string} frameUrl  The URL, including data hash if
+   * applicable, of the cross-domain iframe
+   * @return {!Element}
+   */
+  static createCrossDomainFrame(ampDoc, frameUrl) {
+    // DO NOT MERGE THIS
+    // Warning: the scriptSrc URL below is only temporary. Don't merge
+    // before getting resolution on that.
+    const frame = createElementWithAttributes(ampDoc, 'iframe', {
+      sandbox: 'allow-scripts',
+      name: JSON.stringify({
+        'scriptSrc': '/examples/analytics-3p-remote-frame-helper.js',
+      }),
+    });
+    loadPromise(frame).then(() => {
+      this.setIsReady_(frameUrl);
+    });
+    setStyles(frame,
+        {width: 0, height: 0, display: 'none',
+         position: 'absolute', top: 0, left: 0});
+    Transport.crossDomainFrames[frameUrl] = {
+      frame,
+      isReady: false,
+      msgQueue: [],
+      usageCount: 1
+    };
+    frame.src = frameUrl; // Intentionally doing this after creating load
+    // promise, rather than in the object supplied to
+    // createElementWithAttribute() above. Want to be absolutely
+    // certain that we don't lose the loaded event.
+    return frame;
+  }
+
+  /**
+   * Indicate that a cross-domain frame is ready to receive messages, and
+   * send all messages that were previously queued for it.
+   * @param {!string} frameUrl The URL (including any data hash) of the frame
+   * @private
+   */
+  static setIsReady_(frameUrl) {
+    const frameData = Transport.crossDomainFrames[frameUrl];
+    frameData.isReady = true;
+    this.sendToCrossDomainIframe_(frameData.frame, frameData.msgQueue);
+    delete frameData.msgQueue;
+  }
+
+  /**
+   * Record that one more creative is using this cross-domain iframe
+   * @param {!string} frameUrl The URL (including any data hash) of the frame
+   * @return {number}
+   * @private
+   */
+  static incrementUsageCount_(frameUrl) {
+    return ++(Transport.crossDomainFrames[frameUrl].usageCount);
+  }
+
+  /**
+   * Record that one fewer creative is using this cross-domain iframe
+   * @param {!string} frameUrl The URL (including any data hash) of the frame
+   * @return {number}
+   * @private
+   */
+  static decrementUsageCount_(frameUrl) {
+    return --(Transport.crossDomainFrames[frameUrl].usageCount);
+  }
+
+  /**
+   * Send an array of messages to a cross-domain iframe
+   * @param {!Element} frame  The cross-domain iframe
+   * @param {!Array<string>} messages  The messages to send
+   * @private
+   */
+  static sendToCrossDomainIframe_(frame, messages) {
+    // DO NOT MERGE THIS
+    // Warning: the following code is likely only temporary. Don't check
+    // in before getting resolution on that.
+    frame && frame.contentWindow &&
```

Comment:
Yes, they should...but figured if somehow they don't, the consequences of blasting ahead would be very bad. Will switch to assert() instead.